rust/transaction-filter/src/filters/write_set_change_filter.rs

```diff
@@ -0,0 +1,306 @@
+use crate::{filters::MoveStructTagFilter, traits::Filterable};
+use anyhow::Error;
+use aptos_protos::transaction::v1::{
+    write_set_change::Change, DeleteModule, DeleteResource, DeleteTableItem, WriteModule,
+    WriteResource, WriteSetChange, WriteTableItem,
+};
+use serde::{Deserialize, Serialize};
+
+/// This is a wrapper around ChangeItemFilter, which differs because:
+/// While `ChangeItemFilter` will return false if the Event does not match the filter,
+/// `ChangeItemFilter` will return true- i.e `WriteSetChangeFilter` *only* tries to match if the
+/// change type matches its internal change type
```

Comment:
Do we need both the `ChangeItemFilter` and `WriteSetChangeFilter`? If it's possible, I prefer having a single one to keep things simpler. 

Also how will we check a transaction if a transaction is allowed? Is it if _any_ of its wsc is allowed?

Reply:
currently this `simple` version- yeah

In the follow up PR, there is a `AND`, `OR`, and `NOT`